Hadrons/Modules/MContraction/Meson.hpp

```diff
@@ -131,12 +132,37 @@ void TMeson<FImpl1, FImpl2>::parseGammaString(std::vector<GammaPair> &gammaList)
                 gammaList.push_back(std::make_pair((Gamma::Algebra)i, 
                                                    (Gamma::Algebra)j));
             }
+            gammaSnk[i/2]=1;
         }
     }
     else
     {
         // Parse individual contractions from input string.
-        gammaList = strToVec<GammaPair>(par().gammas);
+        std::vector<GammaPair> tmp;
+        tmp = strToVec<GammaPair>(par().gammas);
+        // assert that only legal Gamma matrices are used
+        for (unsigned int j = 0; j < tmp.size(); j++)
+        {
+            // wrong strings are parsed to undef, equaling (Gamma::Algebra)(-1) as defined in
+            // as defined in Grid/Grid/serialisation/MacroMagic.h, line 161 
+            if( (tmp[j].first==(Gamma::Algebra)(-1)) ||  (tmp[j].second==(Gamma::Algebra)(-1)) )
```

Comment:
don't hardcode `-1`, as in principle this convention could change in the lower-level code. You should be able to use `Gamma::Algebra::undef` instead.